restapi/src/main/java/io/stargate/web/docsapi/service/util/DocsApiUtils.java

```diff
@@ -292,11 +308,12 @@ public static boolean isRowOnPath(Row row, Iterable<String> pathIterable) {
       // if we have the path segment, we need to check if any matches
       if (pathSegment) {
         boolean noneMatch =
-            Arrays.stream(target.split(",")).noneMatch(t -> Objects.equals(t, path));
+            Arrays.stream(target.split(COMMA_PATTERN.pattern()))
+                .noneMatch(t -> Objects.equals(DocsApiUtils.convertEscapedCharacters(t), path));
```

Comment:
ahd here

Reply:
So this case right here is actually the reason for waiting til the "last moment" before query building to do the unescaping...

If you have a value for p0 in the database that is literally `a,b` and you want to match it, you would have to send `{ "a\,b": { "$exists": true }` as your where clause. If you unescape "a\,b" too _early_, then splitting on a comma produces `["a", "b"]` here (which is the "a or b path" syntax that we allow with commas). By keeping the `\` sequence until the last moment, this COMMA_PATTERN (which uses a negative lookback to check for `\`) does NOT split "a\,b" at all, and then only _after_ the attempt to split does unescaping occur.  So instead of producing `["a", "b"]` after the split on this line, you instead get `["a,b"]` which actually matches the value in the database...

There are two potentially bad things here but I couldn't think of another way to do it...one is the COMMA_PATTERN (and also PERIOD_PATTERN) which has a negative lookback, essentially the COMMA_PATTERN regex is "any comma that is not preceded by a `\`"

And the second thing, as you pointed out, is that the unescaping occurs right before querying or matching in memory...any thoughts on how to make this cleaner?

Reply:
@EricBorczuk Understood. I would like us to sync on this and think what's the best solution here?